var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -0,0 +1,93 @@
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Amrex(CMakePackage):
+    """A Block-Structured AMR Framework. AMReX is the successor to BoxLib."""
+
+    homepage = "https://ccse.lbl.gov/AMReX/index.html"
+    url      = "https://github.com/AMReX-Codes/amrex.git"
+
+    version('develop', git='https://github.com/AMReX-Codes/amrex.git', tag='master')
+    version('17.06', git='https://github.com/AMReX-Codes/amrex.git', commit='836d3c7')
+
+    variant('mpi', default=True, description='Enable MPI parallel support')
+# variant('omp', default=False, description='Enable OpenMP parallel support')
+
+    variant('dims',
+        default='3',
+        values=('1', '2', '3'),
+        multi=False,
+        description='Number of spatial dimensions')
+
+    variant('prec',
+        default='DOUBLE',
+        values=('FLOAT', 'DOUBLE'),
+        multi=False,
+        description='Floating point precision')
+
+    variant('fortran', default=True, description='Enable Fortran support')
+    variant('debug', default=False, description='Enable debugging features')
+    variant('particles', default=False, description='Include particle classes in build')
+
+    depends_on('mpi', when='+mpi')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        cmake_args = [
+            '-DENABLE_POSITION_INDEPENDENT_CODE=ON',
+            '-DBL_SPACEDIM:INT=%d' % int(spec.variants['dims'].value),
+            '-DBL_PRECISION:STRING=%s' % spec.variants['prec'].value,
+        ]
+
+        if '+fortran' in spec:
+            cmake_args += [
+                '-DENABLE_FMG=ON',
+                '-DENABLE_FBASELIB=ON'
+            ]
+
+        if '+debug' not in spec:
+            cmake_args += ['-DBL_DEBUG:INT=0']
+
+        if '+particles' in spec:
+            cmake_args += ['-DBL_USE_PARTICLES:INT=1']
+
+        if '~mpi' in spec:
+            cmake_args += ['-DENABLE_MPI:INT=0']
+        else:
+            cmake_args += [
+                '-DCMAKE_C_COMPILER=%s' % spec['mpi'].mpicc,
+                '-DCMAKE_CXX_COMPILER=%s' % spec['mpi'].mpicxx
+            ]
+            if '+fortran' in spec:
+                cmake_args += [
+                    '-DCMAKE_Fortran_COMPILER=%s' % spec['mpi'].mpifc
+                ]
+            else:
+                cmake_args += ['-DENABLE_FORTRAN_MPI=OFF']
```

Comment:
Is it worth pulling this out of the `+fortran` conditional like so?

```python
cmake_args += ['-DENABLE_FORTRAN_MPI=%s' % ('+fortran' in spec)]
```

Then regardless of what the default is for the amrex build, we'll be explicit about whether fortran MPI is on or off.  Otherwise you're relying on the build's default being `ON`.  Note the trick above: CMake will actually understand `True` and `False`, which are the string values of Python's boolean type.